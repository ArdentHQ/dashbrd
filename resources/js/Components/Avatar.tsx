import Jazzicon, { jsNumberForAddress } from "react-jazzicon";

interface Properties {
    address: string;
    size?: number;
    avatar?: App.Data.Wallet.WalletAvatarData;
}

export const Avatar = ({ address, size = 20, avatar }: Properties): JSX.Element => {
    if (avatar?.default !== null && avatar?.default !== undefined) {
        const sourceSet = [
            avatar.small !== null ? `${avatar.small} 1x` : null,
            avatar.small2x !== null ? `${avatar.small2x} 2x` : null,
        ]
            .filter((value) => value !== null)
            .join(", ");

        return (
            <img
                data-testid="Avatar__image"
                src={avatar.default}
                srcSet={sourceSet}
                className="rounded-full"
                style={{ width: size, height: size }}
            />
        );
    }
    return (
        <Jazzicon
            diameter={size}
            seed={jsNumberForAddress(address)}
        />
    );
};
