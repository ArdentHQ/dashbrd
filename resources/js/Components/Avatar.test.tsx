import React from "react";
import { Avatar } from "./Avatar";
import { render, screen } from "@/Tests/testing-library";

describe("Avatar", () => {
    it("should render", () => {
        const { asFragment } = render(<Avatar address="0x00" />);

        expect(asFragment()).toMatchSnapshot();
    });

    it("should render with custom avatar", () => {
        render(
            <Avatar
                address="0x00"
                avatar={{
                    default: "https://dashbrd.test/avatar.png",
                    small: "https://dashbrd.test/small.png",
                    small2x: "https://dashbrd.test/small2x.png",
                }}
            />,
        );

        const img = screen.getByTestId("Avatar__image");

        expect(img).toBeInTheDocument();

        expect(img).toHaveAttribute("src", "https://dashbrd.test/avatar.png");

        expect(img).toHaveAttribute("srcset", "https://dashbrd.test/small.png 1x, https://dashbrd.test/small2x.png 2x");
    });

    it("should render with avatar if no conversions yet", () => {
        render(
            <Avatar
                address="0x00"
                avatar={{
                    default: "https://dashbrd.test/avatar.png",
                    small: null,
                    small2x: null,
                }}
            />,
        );

        const img = screen.getByTestId("Avatar__image");

        expect(img).toBeInTheDocument();

        expect(img).toHaveAttribute("src", "https://dashbrd.test/avatar.png");

        expect(img).toHaveAttribute("srcset", "");
    });
});
