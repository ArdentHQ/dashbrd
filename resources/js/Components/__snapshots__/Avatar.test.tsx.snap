// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Avatar > should render 1`] = `
<DocumentFragment>
  <div
    class="paper"
    style="border-radius: 50px; display: inline-block; margin: 0px; overflow: hidden; padding: 0px; background-color: rgb(21, 148, 242); height: 20px; width: 20px;"
  >
    <svg
      height="20"
      width="20"
      x="0"
      xmlns="http://www.w3.org/2000/svg"
      y="0"
    >
      <rect
        fill="#2462e1"
        height="20"
        rx="0"
        ry="0"
        transform="translate(2.2433313654953237 -3.3339441395209515) rotate(458.4 10 10)"
        width="20"
        x="0"
        y="0"
      />
      <rect
        fill="#f74401"
        height="20"
        rx="0"
        ry="0"
        transform="translate(-9.602162056031382 4.995219772179145) rotate(268.8 10 10)"
        width="20"
        x="0"
        y="0"
      />
      <rect
        fill="#fc7c00"
        height="20"
        rx="0"
        ry="0"
        transform="translate(-5.672658799563389 18.42500992542711) rotate(117.3 10 10)"
        width="20"
        x="0"
        y="0"
      />
      <rect
        fill="#c71448"
        height="20"
        rx="0"
        ry="0"
        transform="translate(-17.239933713789252 15.501698908425618) rotate(280.5 10 10)"
        width="20"
        x="0"
        y="0"
      />
    </svg>
  </div>
</DocumentFragment>
`;
